mohanadarafe/GuideMe
# Feature/google calendar

**In this pull request, you will find the following features:**

1.  Log in/Log out of your google calendar in settings and select a calendar.

2.  View the events of a calendar in course schedule.

3. Go to my next class arrow in course schedule if the user wrote a location for that event in his calendar.

4. When the user log-in to his google account, account will stay active until the user decides to log-out (Persistent even after closing the app).

**Todo next** 📑

- Only show events happening after current date. (working on it while you guys review)
- Restrict the size of the labels if they are too long (title/description of an event/calendar)

**Issues** 🚨🚨🚨

- When testing the app, you might encounter an error when trying to sign in with google. This is going to be fixed by alain who studied redux specifically to fix it. This happens because we sort of leave the app and get back to it when we sign in (related to async storge). For now, if you see the error, just restart the app. it happens half the time you log in. For more details on this please ask @AlainJobs. 
![93009643_217814292843644_806333865911123968_n](https://user-images.githubusercontent.com/39305283/79197495-d01e4a80-7dff-11ea-90c5-b1cba64ab556.jpg)


- This is not to be merged to master right away, as i said alain fixed the async storage issue with redux. We want to see if there are no bugs with this version first.

**To test the app** 🎯

1. Create a new calendar in your google calendar account.
2. Create events and add a description & a location. (The location will be used as a placeholder in doubleSearch)
3. Open settings and sync in with google.
4. You may find a page where it says do you trust the app, just click on advanced options and continue
![93617554_854779998267202_7793870194757074944_n](https://user-images.githubusercontent.com/39305283/79198208-2213a000-7e01-11ea-8511-6d716d2e9850.jpg)

5. Select a calendar in the list bellow
6. Go check your events in course schedule
7. Events with locations will have an arrow next to them to give directions to the class.

**SPECIAL THANKS to @AlainJobs  @csbduzi for helping me alot during this feature😇. I learned a lot from you guys 🙌🙌**
